tt_metal/hw/firmware/src/slave_idle_erisc.cc

```diff
@@ -0,0 +1,85 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+// clang-format off
+#include "risc_common.h"
+#include "noc_overlay_parameters.h"
+#include "noc_nonblocking_api.h"
+#include "dev_msgs.h"
+#include "stream_io_map.h"
+#include "firmware_common.h"
+#include "tools/profiler/kernel_profiler.hpp"
+#include "risc_attribs.h"
+#include "generated_bank_to_noc_coord_mapping.h"
+#include "circular_buffer.h"
+
+#include "debug/waypoint.h"
+#include "debug/dprint.h"
+#include "debug/stack_usage.h"
+// clang-format on
+
+tt_l1_ptr mailboxes_t *const mailboxes = (tt_l1_ptr mailboxes_t *)(MEM_IERISC_MAILBOX_BASE);
+volatile tt_l1_ptr uint8_t *const slave_idle_erisc_run = &mailboxes->slave_sync.ncrisc;
```

Comment:
maybe use a generic name instead of ncrisc or a union?

Reply:
yeah I had a todo for that